server/service/src/name/query.rs

```diff
@@ -20,9 +19,17 @@ pub fn get_names(
     let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
     let repository = NameRepository::new(&ctx.connection);
 
-    let filter = filter
-        .unwrap_or_default()
-        .r#type(NameType::Patient.not_equal_to());
+    let filter = filter.unwrap_or_default();
+
+    let type_filter = match filter.r#type.clone() {
+        Some(filter_input) => EqualFilter::<NameType> {
+            not_equal_to: Some(NameType::Patient),
+            ..filter_input
+        },
+        None => NameType::Patient.not_equal_to(),
+    };
```

Comment:
```suggestion
     let type_filter = match filter.r#type.clone() {
        Some(filter_input) => filter_input, // If we have a Name type filter, use it, trusting the caller.
        None => NameType::Patient.not_equal_to(), // Default to excluding patients from name searches
    };
```

How about just this, keeps it simple? But doesn't solve a security problem if that's the intention...

I'm not 100% if the intention of that filter was to try to prevent this service method being used for querying patients? There is a patient service which I assume we use for actual patient related things...
If so then I think we hard code that as a filter in the repository layer similar to my suggestion branch. And don't even expose the option to get patient to the graphql layer.
@andreievg do you know the intention here?

Reply:
> 'm not 100% if the intention of that filter was to try to prevent this service method being used for querying patients?

I think so, but this should be done in the NameRepository (as overall filter, and similar in Patient repository), looks like there was a refactor to split Name repository at some point (into patient and store/facility), but looks like it was just partially done.

Ideally I would do hard coded filter in the create_filtered_query for both Patient and Name repository. Also would remove is_patient form NameFilterInput and NameNodeType::Patient

> If so then I think we hard code that as a filter in the repository layer similar to my suggestion branch. And don't even expose the option to get patient to the graphql layer.

Yep i think as you've described, but not quite as the suggested branch implementation (i.e. there is Patient repo and patient filter, this should be used in patient queries, and reserve name completely for other types)

Reply:
Ok awesome thanks team - this is feeling like a bigger change than the scope of the issue... I'll just use the `isStore` filter and create a separate issue for this 🙏

Reply:
#4161